tests/modules/lib/cddl-gen/decode/src/main.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <ztest.h>
+#include "test_decode.h"
+#include "cbor_encode.h"
+
+static void test_decode(void)
+{
+	uint8_t payload[32] = {0};
+	uint32_t payload_len;
+	uint8_t time[] = {1, 2, 3, 4, 5, 6, 7, 8};
+	cbor_state_t states[4];
+	struct Test test;
+	bool res;
+
+	new_state(states, 4, payload, sizeof(payload), 1);
+
+	res = list_start_encode(states, 3);
+	res |= list_start_encode(states, 2);
+	res |= tstrx_put_term(states, "Foo");
+	res |= tstrx_put_term(states, "Bar");
+	res |= list_end_encode(states, 0);
+	res |= bstrx_encode(states, &(cbor_string_type_t){.value = time, .len = (sizeof(time))});
+	res |= intx32_put(states, 2);
+	res |= list_end_encode(states, 0);
+	zassert_true(res, "Encoding should have been successful\n");
+
+	uint32_t len = states->payload - payload;
+
+	res = cbor_decode_Test(payload, len, &test, &payload_len);
```

Comment:
Should we (perhaps in the future) verify that the contents of `test` is as expected? Currently we don't check (AFAICS) that the decoding has been done correctly, only that it did not return an error.

Reply:
I left it out since it's extensively tested in the cddl-gen test suite.